design/ceph/ceph-telemetry.md

```diff
@@ -0,0 +1,101 @@
+- Num CSI plugin pods running - This gives idea of how many k8s nodes are in the k8s cluster that
+  are allowed to use Ceph storage.
+  - It is easiest count plugin pods separately since each DaemonSet is a different label. Plus, it
+    can show if some CSI drivers are ever disabled.
+  - `rook-num-csi-rbd-plugin-pods`
+  - `rook-num-csi-cephfs-plugin-pods`
+  - `rook-num-csi-nfs-plugin-pods`
+- `rook-num-nodes` - Number of nodes on which the cluster has running pods.
+  - Since clusters with portable PVCs have one "node" per PVC, this will help show the actual node
+    count for Rook installs in portable environments
+  - We can get this by counting the number of crash collector pods; however, if any users disable
+    the crash collector, we should report an unknown value, possibly `unknown`.
+- `rook-num-storage-classes` - Number of storage classes related to the current Ceph cluster. We can
+  also break this up by specific kinds of storage classes, but the overall may be useful if new
+  classes are added in the future.
+  - `rook-num-rbd-storage-classes`
+  - `rook-num-cephfs-storage-classes`
+  - `rook-num-nfs-storage-classes`
+  - `rook-num-bucket-storage-classes`
+- `rook-num-storage-class-device-sets` - Number of storage class device sets.
+- `rook-num-portable-storage-class-device-sets` - .
+- `rook-mons-are-pvc` - true/false whether mons are on PVC.
+- `rook-network-provider` - The network provider used for the cluster. e.g., default, host, multus.
+- `rook-external-mode` - true/false if the cluster is in external mode.
+
+## Information Rook is interested in that is already collected by Ceph telemetry
+- RBD pools (name is stripped), some config info, and the number of them.
+- Ceph Filesystems and MDS info.
+- RGW count, zonegroups, zones.
+- RBD mirroring info.
+
+## Metrics that Rook would like to see added to Ceph's telemetry
+- Number of NFS clusters
+  - Ceph can count NFS clusters by listing the namespaces in the `.nfs` pool and report those
+    namespaces that have `conf-nfs.<namespace>` objects present
+- Filesystem mirroring info
+- RGW bucket notification usage (Yuval may have good detail on this point)
+
+## Proposed integration strategy
+- Rook will first merge [`rook-version`](https://github.com/rook/rook/pull/10161).
+- Rook, with input from Ceph telemetry, will approve a version of this design doc and treat all
+  noted metric names as requirements.
+- Ceph will add all of the noted telemetry keys from the config-key database to its telemetry and
+  backport to Ceph Pacific and Ceph Quincy.
+  - Ceph code should handle the case where Rook has not set these fields.
+  - Ceph code can add `rook-version` immediately if they expect it might get into the next
+    Pacific/Quincy release more quickly.
+- Rook will implement each metric (or related metric group) individually over time.
+
+This strategy will allow Rook time to add telemetry items as it is able without rushing. Because the
```

Comment:
I wonder if a Ceph tracker will be the best way to signal that we are ready for them to pick up the next batch of rook updates, this would be good to confirm.